src/models/tree/build_tree.h

```diff
@@ -135,7 +134,6 @@ class SparseSplitProposals {
     auto feature_row_end   = row_pointers[feature + 1];
     auto* ptr              = thrust::lower_bound(
       thrust::seq, split_proposals.ptr(feature_row_begin), split_proposals.ptr(feature_row_end), x);
-    if (ptr == split_proposals.ptr(feature_row_end)) { return NOT_FOUND; }
```

Comment:
Would it make sense to throw an exception here?  So that there is no need to check that the user set up with an `inf` value correctly and `nan` values are checked (if they even hit this).

Reply:
Legate-boost does not accept NaN or inf currently and will throw here https://github.com/rapidsai/legate-boost/blob/98412e3f28e4dd493a321b1a8da685d88a788f4c/legateboost/input_validation.py#L52

I think it does make sense to throw an exception. I was a bit on the fence about it because its a hot code path but I think its fine.